helpdesk_mgmt_partner_sequence/models/helpdesk_ticket.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2020 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class HelpdeskTicket(models.Model):
+    _inherit = "helpdesk.ticket"
+
+    def _prepare_ticket_number(self, values):
+        seq = False
+        if "partner_id" in values or self.partner_id:
+            partner = (
+                self.env["res.partner"].browse(values["partner_id"])
+                if "partner_id" in values
+                else self.partner_id
+            )
+            company_id = (
+                values["company_id"] if "company_id" in values else self.company_id.id
+            )
+            # look for ticket sequence in partner
+            seq = partner.helpdesk_ticket_sequence_id
```

Comment:
in this block wouldn't it be better change the partner by commercial_partner_id?

Reply:
I don't understand that suggested change, could you explain?

Reply:
In this case, One contact can be one sequence diferent that the parent company?

Reply:
Yes, that's one of the functional goals:
- Use the contact ticket sequence, if defined. There's a validation if sequence company doesn't match with ticket company, in that case sequence is ignored.
- If the contact hasn't associated sequence, try the same with parent concat (company of the contact).
- Then, it's possible to use a general custom sequence for a contact company and their contacts, and for a certain contact of this company use another specific sequence.

Reply:
This functionality should be explained in the readme file.
hmm, but if I have multiple companies, this functionality doesn't work well, because I can't share the sequence with the other companies. The company_id field in the sequence is not mandatory, I would change the if to:

`if not seq.company_id or seq.company_id.id! = company_id: `

What do you think?

Reply:
The code it's fine ^^, You just need to add the 'functional goals' in the Readme folder.